### ISSUE TITLE
Saved entries disappear from the list immediately after saving and reappaer after reload

### ISSUE THREAD

**Issue by lekkas** [resin-io/pensieve#175]
Opening this as a placeholder, it could be that I'm simply getting an older pensieve.js version (see  https://github.com/resin-io/pensieve/issues/174 )

**Comment by lekkas** [comment]
cc @sonyagreen who I think hit this with Scratchpad as well

**Comment by lekkas** [comment]
While I haven't been able to consistently reproduce the entry disappearing issue described in the title, I've been able to consistently reproduce this which looks related

- Create a new entry
- Save the entry
- The entry will appear, but will disappear on page reload.
- For a few seconds no new entries can be created (`409 - conflict error`)

**Comment by sonyagreen** [comment]
@dfunckt ran into this today. Even refreshing, he didn't see it. When I refreshed, I could see it when he couldn't.

**Comment by LucianBuzzo** [comment]
In theory https://github.com/resin-io/pensieve/pull/187 should help this issue

**Comment by sonyagreen** [comment]
I recreated this today.
* Create a new entry
* Save the entry
* The entry will appear, but will disappear on page reload.
* For a few seconds no new entries can be created - `409 error making request PUT https://api.github.com/repos/resin-io/scratchpad/contents/scratchpad.yaml: "Conflict"`
* Entry showed up again (for good) within 30 seconds.

**Comment by brownjohnf** [comment]
I experienced this today:

* create a new entry. entry appears on the page after save
* reload the page, the entry is missing
* check github; the entry is in the yaml file
* hard refresh the page, the entry reappears

I assume i actually hit this issue, but didn't attempt to create any new entries so I didn't see the 409 error.

**Comment by lurch** [comment]
> The entry will appear, but will disappear on page reload.
Entry showed up again (for good) within 30 seconds.

Kinda sounds like there's a latency issue between writes (to GitHub) and reads (from GitHub)? If the latency can't be solved (GitHub API issues?) perhaps Pensieve could display some kind of "busy" status until it's sure that GitHub is "fully refreshed"?
(this is just my naive un-informed opinion :wink: )

Perhaps @hedss will have more info about / experience with GitHub API latency issues?

**Comment by hedss** [comment]
Yes, this is not a shock to me, Github def. has an issue with latency between a 200 from a POST of data, and being able to retrieve that data.

The best hope you have for this is to request webhook events and then wait for an event informing you that the data's been committed. However, the bad news is I've seen issues where an event's been posted but the data isn't actually available yet. :( The best you can do here is to retry a fetch until you get the data or for a timeout and complain something's up (which is what VersionBot does). Sucks, but there you go. :(
